lib/services/http_service/proposal_model.dart

```diff
@@ -0,0 +1,123 @@
+class ProposalModel {
+  final int id;
+  final String creator;
+  final String recipient;
+  final String quantity;
+  final String staked;
+  final int executed;
+  final int total;
+  final int favour;
+  final int against;
+  final String title;
+  final String summary;
+  final String description;
+  final String image;
+  final String url;
+  final String status;
+  final String stage;
+  final String fund;
+  final int creationDate;
+
+  ProposalModel({
+    this.id,
+    this.creator,
+    this.recipient,
+    this.quantity,
+    this.staked,
+    this.executed,
+    this.total,
+    this.favour,
+    this.against,
+    this.title,
+    this.summary,
+    this.description,
+    this.image,
+    this.url,
+    this.status,
+    this.stage,
+    this.fund,
+    this.creationDate,
+  });
+
+  factory ProposalModel.fromJson(Map<String, dynamic> json) {
+    return ProposalModel(
+      id: json["id"],
+      creator: json["creator"],
+      recipient: json["recipient"],
+      quantity: json["quantity"],
+      staked: json["staked"],
+      executed: json["executed"],
+      total: json["total"],
+      favour: json["favour"],
+      against: json["against"],
+      title: json["title"],
+      summary: json["summary"],
+      description: json["description"],
+      image: json["image"],
+      url: json["url"],
+      status: json["status"],
+      stage: json["stage"],
+      fund: json["fund"],
+      creationDate: json["creation_date"],
+    );
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is ProposalModel &&
+          runtimeType == other.runtimeType &&
+          id == other.id &&
+          creator == other.creator &&
+          recipient == other.recipient &&
+          quantity == other.quantity &&
+          staked == other.staked &&
+          executed == other.executed &&
+          total == other.total &&
+          favour == other.favour &&
+          against == other.against &&
+          title == other.title &&
+          summary == other.summary &&
+          description == other.description &&
+          image == other.image &&
+          url == other.url &&
+          status == other.status &&
+          stage == other.stage &&
+          fund == other.fund &&
+          creationDate == other.creationDate;
+
+  @override
+  int get hashCode =>
```

Comment:
Where is it used?

Reply:
Using it for the hero animation.
I didn't want to use the id because if for whatever reason the id is null for two items this would break the UI.